### PR TITLE
fix: [CI-3950]: Fix supported input types for all list type fields

### PR DESCRIPTION
**Pull request by vardanbansal-harness**
##### Summary:

Fix for https://harness.atlassian.net/browse/CI-3950

As part of this revisited all the fields of type list and fixed the issue wherever applicable.

Also, addressed another design gap where [https://github.com/harness/harness-core-ui/blob/develop/src/modules/10-common/components/MultiTypeList/MultiTypeList.tsx](url) component doesn't handle expression as a value properly. So hiding it from UI till fixed properly.

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx [hunk 1]
```diff
@@ -377,35 +391,62 @@ export const CIStepOptionalConfig: React.FC<CIStepOptionalConfigProps> = props =
...
+                allowedTypesForEntries: SupportedInputTypesForListItems,
+                expressions,
+                getString,
+                readonly,
+                formik
+              })
+            : renderMultiTypeList({
+                name: `${prefix}spec.args`,
+                labelKey: 'argsLabel',
+                tooltipId: 'dependencyArgs',
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx:446]: should this be argsLabel?

File: src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx [hunk 2]
```diff
@@ -377,35 +391,62 @@ export const CIStepOptionalConfig: React.FC<CIStepOptionalConfigProps> = props =
...
-            name: `${prefix}spec.args`,
-            labelKey: 'argsLabel',
-            tooltipId: 'dependencyArgs',
-            allowedTypes: isInputSetView ? AllMultiTypeInputTypesForInputSet : AllMultiTypeInputTypesForStep
-          })}
+          {isInputSetView
+            ? renderMultiTypeListInputSet({
+                name: `${prefix}spec.args`,
+                tooltipId: 'argsLabel',
+                labelKey: 'entryPointLabel',
```
Comment (mtran7) [inline src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx:435]: tooltip & label don't align unless this is intended?

File: src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx:446]: looks like a typo while copy-paste, let me fix it.

**Review by mtran7 — DISMISSED** [review]

File: src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx (hunk 1 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx:446]: Done.

File: src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx (hunk 2 above)
Comment (vardanbansal-harness) [reply src/modules/75-ci/components/PipelineSteps/CIStep/CIStepOptionalConfig.tsx]: Fixed.

**Review by mtran7 — DISMISSED** [review]

**Review by mtran7 — DISMISSED** [review]

**Comment by vardanbansal-harness** [comment]
retrigger jest

**Review by ishantmehtaharness — APPROVED** [review]
LGTM

**Pull request merged**